src/App.tsx

```diff
@@ -1,19 +1,135 @@
 import React from 'react';
-import './App.css';
-
+import './App.scss';
+import { TodoList } from './components/TodoList/TodoList';
+import { todos } from './api/todos';
 import users from './api/users';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+const preparedTodos: Todo[] = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || null,
+}));
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+type State = {
+  todos: Todo[],
+  title: string,
+  user: User | null,
+  selectUserName: string,
+  needShowTitleError: boolean,
+  needShowUserError: boolean,
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    todos: preparedTodos,
+    selectUserName: '',
+    title: '',
+    user: null,
+    needShowTitleError: false,
+    needShowUserError: false,
+  };
+
+  addTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      title: event.target.value,
+      needShowTitleError: false,
+    });
+  };
+
+  addUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    const selectedUser = users.find(user => user.name === value);
+
+    this.setState({
+      selectUserName: value,
+      user: selectedUser || null,
+    });
```

Comment:
maybe we should not add a todo if the user is not found? who will be doing the todo then?